### ISSUE TITLE
Add Report Build Status Argument to CodeBuild Project Resource for Bitbucket

### ISSUE THREAD

**Issue by vibhuyadav** [terraform-providers/terraform-provider-aws#6296]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add AWS Codebuild report build status for Bitbucket

### New or Affected Resource(s)

* aws_codebuild_project

### Potential Terraform Configuration

```hcl
resource "aws_codebuild_project" "test_project"{
        ....
        source {
        ....
            report_build_status=true
        ....
        }
        ....
 }
```

### References

A similar request for Github as the SCM can be found here:
* #5153

**Comment by bflad** [comment]
Pull request submitted (we are currently filtering `report_build_status` by `type` to prevent errors): #6426

**Comment by bflad** [comment]
Support has been tested, merged, and will release with version 1.44.0 of the AWS provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.44.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1440-november-14-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
